core/src/types.ts

```diff
@@ -152,3 +152,27 @@ export declare interface CollateralStatus {
     address?: string | null;
     balance?: BigNumber;
 }
+
+export type SurplusAuctionStates = 'just-started' | 'have-bids' | 'ready-for-collection' | 'requires-restart';
+
+export declare interface SurplusAuctionBaseData {
+    id: number;
+    network: string;
+}
+
+export declare interface SurplusAuctionFullData extends SurplusAuctionBaseData {
+    bidAmountMKR: BigNumber;
+    receiveAmountDAI: BigNumber;
+    receiverAddress: string;
+    auctionEndDate: Date;
+    bidEndDate: Date;
+    earliestEndDate: Date;
+}
+
+export declare interface SurplusAuction extends SurplusAuctionFullData {
+    state: SurplusAuctionStates;
+}
```

Comment:
Should `SurplusAuction` be `SurplusAuctionBase | SurplusAuctionCollected`?

Reply:
I do not understand what you're suggesting here exactly.

Reply:
I suggest to have one main `SurplusAuction` type that is a tuple of all other types:
```ts
export declare interface SurplusAuctionBase {
    id: number;
    network: string;
}

export declare interface SurplusAuctionCollected extends SurplusAuctionBase {
    state: 'collected';
}

export declare interface SurplusAuctionActive extends SurplusAuctionBase {
    state: 'just-started' | 'have-bids' | 'ready-for-collection' | 'requires-restart';
    bidAmountMKR: BigNumber;
    receiveAmountDAI: BigNumber;
    receiverAddress: string;
    auctionEndDate: Date;
    bidEndDate: Date;
    earliestEndDate: Date;
}

export type SurplusAuction = SurplusAuctionActive | SurplusAuctionCollected;

export type SurplusAuctionStates = SurplusAuction['state'];
```
This will 
- let us use the same type everywhere and let typescript make most of the work (check that if you access `receiverAddress`, you had checked that `state` is not `collected`)
- also make types directly reusable (you can use `SurplusAuctionCollected` in places where you don't expect generic auction)

Reply:
This comment hasn't been addressed (despite being liked)

Reply:
oops, wrong branch, cherry picked.